conda_forge_tick/migrators/arch.py

```diff
@@ -139,3 +139,132 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
 
     def remote_branch(self, feedstock_ctx: FeedstockContext) -> str:
         return super().remote_branch(feedstock_ctx) + "_arch"
+
+
+class OSXArm(GraphMigrator):
+    """
+    A Migrator that add arm osx builds to feedstocks
+    """
+
+    migrator_version = 1
+    rerender = True
+    # We purposefully don't want to bump build number for this migrator
+    bump_number = 0
+
+    ignored_packages = {"gfortran"}
+    excluded_dependencies = {"fortran_compiler_stub"}
+
+    arches = ["osx_arm64"]
+
+    additional_keys = {
+        "build_platform": {"osx_arm64": "osx_64"},
+        "test_on_native_only": True,
+    }
+
+    def __init__(
+        self, graph: nx.DiGraph = None, name: Optional[str] = None, pr_limit: int = 0,
+    ):
+        # rebuild the graph to only use edges from the arm osx requirements
+        graph2 = nx.create_empty_copy(graph)
+        for node, attrs in graph.nodes(data="payload"):
```

Comment:
This block has removed edges from build deps so it will miss the fortran compiler stubs.